Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -1276,6 +1276,22 @@ static void emitArrayLength(FunctionDefinitionWriter* writer, AST::CallExpressio
     writer->stringBuilder().append(".size()");
 }
 
+static void emitDistance(FunctionDefinitionWriter* writer, AST::CallExpression& call)
+{
+    auto* argumentType = call.arguments()[0].inferredType();
+    if (std::holds_alternative<Types::Primitive>(*argumentType)) {
+        writer->stringBuilder().append("(");
+        writer->visit(call.arguments()[0]);
```

Comment:
does there need to be a call to `abs(` here or is `distance(` supposed to return a signed number?

Reply:
Oops, good catch, I missed that in the spec. I also updated the constant `length` function which had the same issue.